crimson: add support for ms_bind_retry_{delay,count}

rzarzynski opened the pull request:
<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

cyx1231st commented on src/crimson/net/Socket.h:
```diff
@@ -160,6 +160,11 @@ class Socket
 
+using listen_ertr = crimson::errorator<
+  crimson::ct_error::address_in_use, // The address is already bound
+  crimson::ct_error::address_not_available // https://techoverflow.net/2021/08/06/how-i-fixed-python-oserror-errno-99-cannot-assign-requested-address/The address is already bound
```
Remove the trailing "The address is already bound"

cyx1231st commented on src/crimson/net/SocketMessenger.cc:
```diff
@@ -102,27 +102,29 @@ SocketMessenger::try_bind(const entity_addrvec_t& addrs,
+        return bind_ertr::now(); // success!
+      } else if (e == std::errc::address_in_use) {
         return crimson::ct_error::address_in_use::make();
+      } else if (e == std::errc::address_in_use) {
```
`std::errc::address_not_available`?

cyx1231st commented on src/crimson/net/SocketMessenger.cc:
```diff
@@ -126,6 +118,53 @@ SocketMessenger::try_bind(const entity_addrvec_t& addrs,
+        return bind_ertr::now(); // success!
+      } else if (error == std::errc::address_in_use) {
+        return crimson::ct_error::address_in_use::make();
+      } else if (error == std::errc::address_in_use) {
```
`std::errc::address_not_available`?

cyx1231st reviewed (commented):
Make check does not pass.

rzarzynski commented:
Thanks for the review, @cyx1231st! Addressed the comments. The build problem was:

```
../src/crimson/net/SocketMessenger.cc:159:14: error: lambda capture 'this' is not used [-Werror,-Wunused-lambda-capture]
    }).then([this] (const std::error_code error) -> bind_ertr::future<> {
```

So `-Werror` & Clang are doing their work! :-)

cyx1231st commented:
`../src/test/crimson/test_socket.cc:78:12: error: no member named 'listen_ertr' in 'crimson::net::FixedCPUServerSocket'; did you mean 'crimson::net::listen_ertr'?`

rzarzynski commented:
@cyx1231st: fixed the unit test.

cyx1231st reviewed (approved):
LGTM!
